### PR TITLE
Add MongoDB v1.0.1 for Flink 1.16.x and Flink 1.17.x

### PR DESCRIPTION
**Pull request by dannycranmer**

### CODE REVIEW

File: docs/data/release_archive.yml [hunk 1]
```diff
@@ -428,6 +428,11 @@ release_archive:
       version: 1.0.0
       release_date: 2022-12-21
       filename: "opensearch"
+    - name: "Flink MongoDB Connector"
```
Comment (elphastori) [inline docs/data/release_archive.yml:431]: Consider adding version 1.0.0 which seems to not have been listed in the last release.

**Review by elphastori — APPROVED** [review]
Local build looks good

**Review by hlteoh37 — APPROVED** [review]

File: docs/data/release_archive.yml (hunk 1 above)
Comment (MartijnVisser) [reply docs/data/release_archive.yml:431]: +1

**Review by MartijnVisser — APPROVED** [review]

**Review by leonardBang — APPROVED** [review]
LGTM

**Review by vahmed-hamdy — APPROVED** [review]

**Pull request merged**